index.html

```diff
@@ -1884,50 +1909,72 @@ <h2>The <dfn>NDEFRecordType</dfn> string</h2>
       <td><dfn>"`empty`"</dfn></td>
       <td><i>unused</i></td>
       <td><i>unused</i></td>
-      <td><a>Empty record</a></td>
+      <td><a>Empty record</a> (TNF=0)</td>
     </tr>
     <tr>
       <td><dfn>"`text`"</dfn></td>
       <td><i>unused</i></td>
       <td>{{DOMString}}</td>
-      <td><a>Well-known type</a> record with type "`T`"</td>
+      <td><a>Well-known type</a> (TNF=1) record with TYPE="`T`"</td>
     </tr>
     <tr>
       <td><dfn>"`url`"</dfn></td>
       <td><i>unused</i></td>
       <td>{{DOMString}}</td>
-      <td><a>Well-known type</a> record with type "`U`"</td>
+      <td><a>Well-known type</a> (TNF=1) record with TYPE="`U`"</td>
     </tr>
     <tr>
       <td><dfn>"`smart-poster`"</dfn></td>
       <td><i>unused</i></td>
       <td>{{NDEFMessage}}</td>
-      <td><a>Well-known type</a> record with type "`Sp`"</td>
+      <td><a>Well-known type</a> (TNF=1) record with TYPE="`Sp`"</td>
+    </tr>
+    <tr>
+      <td><dfn>"`absolute-url`"</dfn></td>
+      <td><i>unused</i></td>
+      <td>{{DOMString}} <i>url</i></td>
+      <td><a>Absolute-URL record</a> (TNF=2) with TYPE=<i>url</i></td>
     </tr>
     <tr>
       <td><dfn>"`json`"</dfn></td>
       <td><a>JSON MIME type</a></td>
       <td>
         [= JSON type =]
       </td>
-      <td><a>MIME type record</a> with type equal to
-        <a>MIME type</a>.
+      <td><a>MIME type record</a> (TNF=2) with TYPE= [= JSON type =].
       </td>
     </tr>
     <tr>
       <td><dfn>"`opaque`"</dfn></td>
-      <td><a>MIME type</a></td>
+      <td>[= MIME type =]</a></td>
       <td>{{ArrayBuffer}}</td>
-      <td><a>MIME type record</a></td>
+      <td><a>MIME type record</a> (TNF=3) with TYPE= [= MIME type =]</td>
     </tr>
     <tr>
-      <td><a>external type</a></td>
+      <td><i>[= external type =]</i></td>
+      <td><i>unused</i></td>
+      <td>
+        {{ArrayBuffer}} or any<br>
+        <a>typed array type</a> or<br>
+        {{NDEFMessage}}
+      </td>
+      <td><a>External type record</a> (TNF=4) with TYPE= [= external type =]</td>
+    </tr>
+    <tr>
+      <td><i>[= local type =]</i></td>
       <td><i>unused</i></td>
       <td>
-        {{ArrayBuffer}} or<br>
-        any <a>typed array type</a>
+        {{ArrayBuffer}} or any<br>
+        <a>typed array type</a>
       </td>
-      <td><a>External type record</a></td>
+      <td><a>External type record</a> (TNF=4) with TYPE= [= local type =] inside
+        the <a>NDEFMessage</a> payload of another record</td>
+    </tr>
+    <tr>
+      <td><dfn>"`unknown`"</dfn></td>
+      <td><i>unused</i></td>
+      <td>{{ArrayBuffer}}</td>
```

Comment:
Ditto. Are we sure an unknown record could have a ndef message as payload?

Reply:
Those could have anything (binary data), and on the reading side a swiss army knife of trying to make sense out of it with different getters might make sense. But strictly speaking, we could say that only the ArrayBuffer getter is guaranteed to work.

On the creation side, apps can put any payload, including a full message, but I don't see any point in creating "unknown" records from apps. Since we need to deal with "unknown" records on the reading side, for symmetry I included them in the creation side as well, even though it makes little to no practical sense.

Reply:
We were pointed out that all these getters should just throw if they make no sense, and never return `null` (unless it is a valid return type, like if I stored `null` as JSON)

Reply:
In a sense, it is also totally allowed to write an NDEFMessage as opaque data, you just need to invent your own mimetype (say application/ndef) or you will have trouble handling it.

Reply:
Right, those would work, even though one would need to register that MIME type if they want all parsers  handle it.